### PR TITLE
let admission plugins indicate they want nothing

### PR DESCRIPTION
**Pull request by deads2k**
An admission plugin can return `nil, nil` for construction.  This is useful for dealing with cases where the `config` passed to you effectively means, "no work".  The calling code already handles this.

@derekwaynecarr

### CODE REVIEW

File: pkg/admission/plugins.go [hunk 1]
```diff
@@ -63,18 +63,19 @@ func RegisterPlugin(name string, plugin Factory) {
 	plugins[name] = plugin
 }
 
-// GetPlugin creates an instance of the named plugin, or nil if the name is not
+// getPlugin creates an instance of the named plugin, or nil if the name is not
 // known. The error is returned only when the named provider was known but failed
```
Comment (liggitt) [inline pkg/admission/plugins.go]: update doc

**Comment by liggitt** [comment]
nit, LGTM otherwise

**Comment by derekwaynecarr** [comment]
Who was returning `nil, nil` for construction?

**Comment by deads2k** [comment]
> Who was returning nil, nil for construction?

Downstream admission.  Because the chain is highly sensitive to ordering and that ordering is difficult to describe, its safer and easier for users to choose an ordering that works and then admission plugins can check the config to see if they should be doing something.  That lets them turned on via config and always operate at the "right" spot in the chain.

**Comment by deads2k** [comment]
@liggitt  Tagged to play with the bot and new green button behavior.  The queue is red, so this won't go anywhere.

**Comment by deads2k** [comment]
@liggitt do you have a green button on this pull?  Looks like we might finally have separate tagging and assignment permissions.

**Comment by deads2k** [comment]
@eparis @kubernetes/rh-cluster-infra Any idea how I'm able to label and assign, but not merge (no green button)?  Not a complaint.  If there's a more granular permission somewhere, that'd be useful to know.

**Comment by deads2k** [comment]
Looks like dan got it sorted.

@liggitt still need a review

**Pull request merged**